src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -257,10 +257,10 @@ void Server::registerPath(const u16string& path) {
     watchRoots[watchPoint.watchDescriptor] = path;
 }
 
-void Server::unregisterPath(const u16string path) {
+void Server::unregisterPath(const u16string& path) {
     auto it = watchPoints.find(path);
     if (it == watchPoints.end() || it->second.status == FINISHED) {
-        logToJava(FINE, "Path is not watched: %s", utf16ToUtf8String(path).c_str());
+        logToJava(WARNING, "Path is not watched: %s", utf16ToUtf8String(path).c_str());
```

Comment:
I am not sure this should be a warning. That would require the client to not unregister a deleted directory to avoid the warning.

Reply:
I think it makes sense that this is a `WARNING` in the context of the native file watching. The application can decide to disable it if it doesn't care about recoverable issues like this—that's pretty much what we are planning to do in `gradle/gradle`. Another app might care about these warnings, and leave the log level on `INFO`.

Another way to address this is use `INFO` for recoverable issues like this, and only use `WARNING` / `SEVERE` for non-recoverable stuff. This would be a bit strange for me, but acceptable. We should fail the tests then if `INFO` logs appear.

Reply:
We currently can't switch off logging per logger from the Gradle side. I would prefer if we switch to `INFO` logging for recoverable things.

Reply:
We can do it for JUL loggers. But I'm fine with using `INFO`, too.

Reply:
I'm now asserting all log messages >= `INFO`.

I'm also returning the status of `stopWatching()` and asserting that it actually stopped the watchers.